### ISSUE TITLE
ansible_become_pass does not work anymore

### ISSUE THREAD

**Issue by prahal** [debops/debops#2256]
For monthes I have been unable to run debops against my debops controller. I went to believe that the issue is with ansible_become_pass not being sent anymore. Maybe related to the new debops run syntax.

the playbooks hangs doing what seems to be waiting for the sudo password.
```
TASK [Gathering Facts] ***********************************************************************************************************************************************************************
task path: /home/prahal/Projects/Admin/ansible/debops/ansible/playbooks/sys/mount.yml:6
<cyclope> Attempting python interpreter discovery
<cyclope> ESTABLISH LOCAL CONNECTION FOR USER: prahal
<cyclope> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.9'"'"'; command -v '"'"'python3.8'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<cyclope> EXEC /bin/sh -c '/usr/bin/python3.9 && sleep 0'
Using module file /usr/lib/python3/dist-packages/ansible/modules/setup.py
Pipelining is enabled.
<cyclope> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=kokcvkoppopmuaupfmusrvkhqrpxekrd] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-kokcvkoppopmuaupfmusrvkhqrpxekrd ; /usr/bin/python3'"'"' && sleep 0'
```

I find it right that controller requires a sudo password. But I am stuck because of the above issue.

in the inventory `ansible/inventory/hosts` I have 
```
cyclope ansible_connection=local ansible_become_password='{{ phn_cyclope__become_pwd }}'
```
in `ansible/inventory/group_vars/all/vars` I have `phn_cyclope__become_pwd: '{{ vault__phn_cyclope__become_pwd }}'
`and the controller sudo password is correctly defined in `ansible/inventory/group_vars/all/vault`.
This was working a long time ago.

**Comment by prahal** [comment]
I can workaround this issue by disabling ssh pipelining.
That is in `.debops.cf` I have to comment:
```
[ansible ssh_connection]
pipelining=True
```
Still inbvestigating.

**Comment by prahal** [comment]
I have on the controller openssh-server 1:8.4p1-5+deb11u1 , sudo 1.9.5p2-3, openssh-client 1:8.4p1-5+deb11u1 (all bullseye)..
But from the above logs I see that sudo is run with `-u root`. Is it expected? Shouldn't it be as one of my admin_users (root is not in my /etc/ansible/facts.d/core.fact admin_users)? Maybe root is fine as without pipelining it works.

**Comment by drybjed** [comment]
How do you execute Ansible or DebOps command? How did you execute them before when they worked?

From the inventory I assume that you want to execute playbooks locally. I don't use `ansible_become_password` so I'm not sure if the `--ask-become-pass` or `-K` parameter is needed by `ansible-playbook` for it to wait for `sudo` and pass the password along, even if it's specified via a variable. You could try running the command with `-K` to see if it changes anything.

Since you are using `ansible_connection=local` for this host, SSH pipelining shouldn't be relevant. Are you able to run Ansible modules directly? For example, does
```bash
$ ansible cyclope -m setup -b -K
```
command executed in the project directory result in a successful dump of host facts?

**Comment by prahal** [comment]
I execute debops with`debops run site.yml -l cyclope`. Last time it worked I believe the syntax was different ie `debops -l cyclope`  but it has been a long time ago (maybe a year) and it might be another component fault (ie ansible).

The variable did the job before but I am pretty confident I did a few run with `-K` and it worked well too back then.

What is weird is I have ansible_connection=local for this host but if I do not disable ssh_connection pipelining=True, even your command stalls (ie `ansible cyclope -m setup -b -K -vvvv` in the project directory hangs on `<cyclope> EXEC /bin/sh -c 'sudo -H -S  -p "[sudo via ansible, key=ewbvkpdbnweyejsxubihjwdkuqsmehwl] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-ewbvkpdbnweyejsxubihjwdkuqsmehwl ; /usr/bin/python3'"'"' && sleep 0'`).
I guess this makes it an ansible issue or even a sudo one (I have !requiretty in /etc/sudoers.d/system_groups-admins for the admins group and my user is in this group).
I don't know much about ansible_connection local, maybe the fact that ssh pipelining is applied to it explains this issue.

At least now I have a workaround, ie disable ssh pipelining.

**Comment by prahal** [comment]
I ran a debops without pipelining then did a run with debops.cfg with added:
```
ssh_args = -o ControlMaster=auto -o ControlPersist=600s
pipelining=True
control_path_dir = ~/.ansible/cp
control_path = %%(directory)s/%%%%C
```
this with a rebased debops code repository and debops project refresh.

No more issues.

I remove the above changes leaving only `pipelining=True` under `[ansible ssh_connection]`, and still no issue.
So I believe the non-pipelining run fixed something.
Closing as unreproducible.

Thank you for the small ansible test case.

The /etc/sudoers.d/system_groups-admins and well none of the sudoers files were updated by the previous run. Weird. (ie not a missing !requiretty in sudoers issue).
